File: src/event/impl/server/ServerEvent.php
<?php

namespace pocketcloud\cloud\event\impl\server;

use pocketcloud\cloud\event\Event;
use pocketcloud\cloud\server\CloudServer;

abstract class ServerEvent extends Event {

    public function __construct(private readonly CloudServer $server) {}

    public function getServer(): CloudServer {
        return $this->server;
    }
}

File: src/main/java/de/pocketcloud/cloud/event/impl/server/ServerEvent.java
package de.pocketcloud.cloud.event.impl.server;

import de.pocketcloud.cloud.event.Event;
import de.pocketcloud.cloud.server.CloudServer;
import lombok.Getter;

public abstract class ServerEvent extends Event {

    @Getter
    private final CloudServer server;

    public ServerEvent(CloudServer server) {
        this.server = server;
    }
}